regparser/tree/xml_parser/reg_text.py

```diff
@@ -15,20 +15,30 @@
 
 def get_reg_part(reg_doc):
     """
-    The CFR Part number for a regulation is contained within
-    an EAR tag, for a Federal Register notice it's in a REGTEXT tag. Get the
-    part number of the regulation.
+    Depending on source, the CFR part number exists in different places. Fetch
+    it, wherever it is.
     """
 
-    #FR notice
-    reg_text_xml = reg_doc.xpath('//REGTEXT')
-    if reg_text_xml:
-        return reg_text_xml[0].attrib['PART']
-
-    #e-CFR XML
-    reg_ear = reg_doc.xpath('//PART/EAR')
-    if reg_ear:
-        return reg_ear[0].text.split('Pt.')[1].strip()
+    potential_parts = []
+    potential_parts.extend(
+        # FR notice
+        node.attrib['PART'] for node in reg_doc.xpath('//REGTEXT'))
+    potential_parts.extend(
+        # e-CFR XML, under PART/EAR
+        node.text.replace('Pt.', '').strip()
+        for node in reg_doc.xpath('//PART/EAR')
+        if 'Pt.' in node.text)
+    potential_parts.extend(
+        # e-CFR XML, under FDSYS/HEADING
+        node.text.replace('PART', '').strip()
+        for node in reg_doc.xpath('//FDSYS/HEADING')
+        if 'PART' in node.text)
+    potential_parts.extend(
+        # e-CFR XML, under FDSYS/GRANULENUM
+        node.text.strip() for node in reg_doc.xpath('//FDSYS/GRANULENUM'))
+
+    if potential_parts:
+        return potential_parts[0]
```

Comment:
Is there a chance that we need to look for the first non-empty string? Instead of just the first element? I don't think so - but I wanted to check with you.